features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml

```diff
@@ -42,6 +42,15 @@
             <OperationDataCleanUp>
                 <Enable>true</Enable>
             </OperationDataCleanUp>
+            <TempDataCleanup>
+                <!-- Enabling separated cleanup for temporary authentication context data -->
+                <Enable>true</Enable>
+                <!-- When PoolZize > 0, temporary data which have no usage after the authentication flow will be deleted immediately -->
```

Comment:
Data will be deleted immediately during the flow when pool size is 0 right

Reply:
No.
If the PoolSize is 0, data will not be deleted until the cleanup task